pkg/solana/chainreader/event_read_binding.go

```diff
@@ -262,3 +384,25 @@ func (r remapHelper) remapExpression(key string, expression query.Expression) (q
 
 	return r.primitive(expression)
 }
+
+type indexedSubkeys struct {
+	lookup  map[string]uint64
+	subKeys [4][]string
+}
+
+func newIndexedSubkeys() *indexedSubkeys {
+	return &indexedSubkeys{
+		lookup: make(map[string]uint64),
+	}
+}
+
+func (k *indexedSubkeys) addForIndex(offChainPath, onChainPath string, idx uint64) {
+	k.lookup[offChainPath] = idx
+	k.subKeys[idx] = strings.Split(onChainPath, ".")
+}
+
+func (k *indexedSubkeys) indexForKey(key string) (uint64, bool) {
+	idx, ok := k.lookup[key]
+
+	return idx, ok
```

Comment:
```suggestion
	return k.lookup[key]
```

Reply:
This hasn't worked in the past. Does it work now?

Reply:
Nope. Doing a direct return like that assumes the value without the boolean check. :(